detect_pcap_version.c should be a C++ file

### Bug description

When trying to find_package PcapPlusPlus installed with vcpkg, it fails with the following:
```
CMake Error at build/vcpkg_installed/x64-linux/share/pcapplusplus/FindPCAP.cmake:113 (try_run):
  Unknown extension ".c" for file

    .../build/detect_pcap_version.c

  try_compile() works only for enabled languages.  Currently these are:

    CUDA CXX

  See project() command to enable other languages.
```

My project indeed only enable those two languages and not C.
It used to work but block my upgrade to PcapPlusPlus 24.09.

I think `detect_pcap_version.c` (introduced in 71e19fe1fbd2cc05094a152679ba4e4aa39b0aa9) should be a C++ file so it doesn't require C to be enabled.

### PcapPlusPlus versions tested on

Other (specify in the next question)

### Other PcapPlusPlus version (if applicable)

24.09

### Operating systems tested on

Linux

### Other operation systems (if applicable)

_No response_

### Compiler version

gcc 13.2.0

### Packet capture backend (if applicable)

_No response_

--- a/cmake/modules/FindPCAP.cmake
+++ b/cmake/modules/FindPCAP.cmake
@@ -90,8 +90,8 @@ if(HAVE_PCAP_LIB_VERSION AND NOT CMAKE_CROSSCOMPILING)
   # Simple C code to extract the libpcap version
   set(PCAP_VERSION_CODE
       "
-  #include <stdio.h>
-  #include <string.h>
+  #include <cstdio>
+  #include <cstring>
   #include <pcap/pcap.h>
 
   int main() {
@@ -106,7 +106,7 @@ if(HAVE_PCAP_LIB_VERSION AND NOT CMAKE_CROSSCOMPILING)
   ")
 
   # Write the code to a temporary file
-  set(detect_pcap_version_file "${PROJECT_BINARY_DIR}/detect_pcap_version.c")
+  set(detect_pcap_version_file "${PROJECT_BINARY_DIR}/detect_pcap_version.cpp")
   file(WRITE "${detect_pcap_version_file}" "${PCAP_VERSION_CODE}")
 
   # Try to compile and run the program

